### PR TITLE
Fix hmr for react components

### PR DESCRIPTION
**Pull request by GiladShoham**
## Proposed Changes

- first phase of fixing HMR using `@pmmmwh/react-refresh-webpack-plugin`
- add an option for pkg getModulePath to get it as an absolute path
- upgrade webpack dev server version
- add react-refresh

Left todo:
- there is a bug that causes the first composition of a component to make a full refresh when updated
- enable the hmr for UI server

### CODE REVIEW

**Review by guysaar223 — APPROVED** [review]

File: scopes/ui-foundation/ui/webpack/webpack.dev.config.ts [hunk 1]
```diff
@@ -318,10 +318,10 @@ function createWebpackConfig(workspaceDir, entryFiles, title, aspectPaths): Webp
...
 
     plugins: [
-      new ReactRefreshWebpackPlugin({
-        // exclude: /@pmmmwh/, // replaces the default value of `/node_modules/`
-        include: aspectPaths,
-      }),
+      // new ReactRefreshWebpackPlugin({
+      //   // exclude: /@pmmmwh/, // replaces the default value of `/node_modules/`
+      //   include: aspectPaths,
+      // }),
```
Comment (KutnerUri) [inline scopes/ui-foundation/ui/webpack/webpack.dev.config.ts]: why this?

File: scopes/react/react/webpack/webpack.config.component.dev.ts [hunk 2]
```diff
@@ -72,24 +73,25 @@ export default function (workDir: string): Configuration {
...
+          // TODO: check why webpackHotDevClient and react-error-overlay are not responding for runtime
+          // errors
+          entry: require.resolve('./react-hot-dev-client'),
+          module: require.resolve('./refresh'),
+        },
+        include: componentsDirs,
+        // TODO: use a more specific exclude for our selfs
+        // exclude: [/dist/, /node_modules/],
+        // exclude: new RegExp('node_modules/\@pmmmwh', 'i'),
+        exclude: '/react-refresh-webpack-plugin/i',
```
[the remark this answers is not in the document]
Comment (KutnerUri) [reply scopes/react/react/webpack/webpack.config.component.dev.ts:91]: please add a comment: 
```suggestion
        // set a value, or else the plugin defaults to ignoring `/node_modules/`
        exclude: '/react-refresh-webpack-plugin/i',
```

File: scopes/react/react/webpack/webpack.config.component.dev.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
 import path from 'path';
-import { Configuration } from 'webpack';
+import webpack, { Configuration } from 'webpack';
```
Comment (KutnerUri) [inline scopes/react/react/webpack/webpack.config.component.dev.ts]:
```suggestion
import type { Configuration } from 'webpack';
```

Comment (KutnerUri) [reply scopes/react/react/webpack/webpack.config.component.dev.ts]: webpack doesn't seem used

File: scopes/pkg/pkg/pkg.main.runtime.ts [hunk 4]
```diff
@@ -178,9 +178,13 @@ export class PkgMain {
...
    * returns the package path in the /node_modules/ folder
    */
-  getModulePath(component: Component) {
+  getModulePath(component: Component, options: { absPath: boolean }) {
     const pkgName = this.getPackageName(component);
-    return join('node_modules', pkgName);
+    const relativePath = join('node_modules', pkgName);
+    if (options?.absPath && this.workspace) {
+      return join(this.workspace.path, relativePath);
+    }
```
Comment (KutnerUri) [inline scopes/pkg/pkg/pkg.main.runtime.ts:186]: love it, it's growing and providing more value. Won't this break on scope though? someone sometime will use it from a scope

**Review by KutnerUri — APPROVED** [review]

**Pull request merged**